public/jsx/views/card-details.jsx

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+var $ = require('jquery');
+var React = require('react');
+
+var CardForm = require('components/card-form');
+var Spinner = require('components/spinner');
+
+
+module.exports = React.createClass({
+
+  displayName: 'CardDetailsView',
+
+  getInitialState: function() {
+    return {
+      braintree_token: false,
+    };
+  },
+
+  propTypes: {
+    apiSource: React.PropTypes.string.isRequired,
+  },
+
+  getDefaultProps: function() {
+    return {
+      apiSource: '/api/braintree/token/generate/',
+    };
+  },
+
+  componentDidMount: function() {
+    $.ajax({
+      method: 'post',
+      url: this.props.apiSource,
+      context: this,
+    }).then(function(data) {
+      if (this.isMounted()) {
+        this.setState({'braintree_token': data.token}); // eslint-disable-line
```

Comment:
It's not that - setting state in componetDidMount is picked up by the react-eslint-plugin as this: https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-did-mount-set-state.md

But in our case we're ok because thrashing won't happening + this is following the way the React docs say it should be done.